app/controllers/admin/routes.py

```diff
@@ -141,19 +143,43 @@ def editEvent(eventId):
     userHasRSVPed = EventRsvp.get_or_none(EventRsvp.user == g.current_user, EventRsvp.event == event)
     isPastEvent = (datetime.now() >= datetime.combine(event.startDate, event.timeStart))
     program = event.singleProgram
-    isProgramManager = hasPrivilege(g.current_user,program)
 
-    if (g.current_user.isStudent or g.current_user.isFaculty) and not isProgramManager:  # only formats to 12 hour time if user doesn't have access (for display purposes)
+    rule = request.url_rule
+    if 'edit' in rule.rule:
+        isProgramManager = hasPrivilege(g.current_user,program)
+        return render_template("admin/createSingleEvent.html",
+                                eventData = eventData,
+                                allFacilitators = getAllFacilitators(),
+                                futureTerms=futureTerms,
+                                isPastEvent = isPastEvent,
+                                userHasRSVPed = userHasRSVPed,
+                                isProgramManager = isProgramManager)
+    else:
+        eventFacilitators = EventFacilitator.select().where(EventFacilitator.event == eventId)
+        eventFacilitatorNames = [eventFacilitator.user for eventFacilitator in eventFacilitators]
         eventData['timeStart'] = datetime.strptime(eventData['timeStart'], "%H:%M").strftime("%I:%M %p")
         eventData['timeEnd'] = datetime.strptime(eventData['timeEnd'], "%H:%M").strftime("%I:%M %p")
-
-    return render_template("admin/createSingleEvent.html",
-                            eventData = eventData,
-                            allFacilitators = getAllFacilitators(),
-                            futureTerms = futureTerms,
-                            isPastEvent = isPastEvent,
-                            userHasRSVPed = userHasRSVPed,
-                            isProgramManager = isProgramManager)
+        eventData["startDate"] = eventData["startDate"].strftime("%m/%d/%Y")
+        programManager = ProgramManager.get_or_none(program=program)
+        programTrainings = Event.select().join(ProgramEvent).where(Event.isTraining == 1, ProgramEvent.program == program)
+        listOfProgramTrainings = [programTraining for programTraining in programTrainings]
+        userParticipatedEvents = {}
+        for training in listOfProgramTrainings:
+            eventParticipants = getEventParticipants(training.id)
+            if training.startDate>date.today():
+                didParticipate = None
+            elif g.current_user.username in eventParticipants.keys():
+                didParticipate = True
+            else:
+                didParticipate = False
+            userParticipatedEvents[training.name] = didParticipate
```

Comment:
This logic to get the list of trainings participated in should be its own function

Reply:
We should combine the render_template calls, so we don't accidentally break one view by forgetting to add a keyword in two places.

Reply:
resolved

Reply:
working on test suite now

Reply:
resolved